components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/ApplicationAuthenticatorService.java

```diff
@@ -97,10 +100,35 @@ public List<LocalAuthenticatorConfig> getAllLocalAuthenticators(String tenantDom
             throws AuthenticatorMgtException {
 
         List<LocalAuthenticatorConfig> configList = new ArrayList<>(getAllUserDefinedLocalAuthenticators(tenantDomain));
-        configList.addAll(localAuthenticators);
+        List<LocalAuthenticatorConfig> systemLocalAuthenticators = getAllSystemDefinedLocalAuthenticators(tenantDomain);
+        configList.addAll(resultantSystemDefinedAuthenticators(systemLocalAuthenticators));
         return configList;
     }
 
+    //Merge the 'localAuthenticators' and the system defined local authenticators already persisted on the database.
+    private List<LocalAuthenticatorConfig> resultantSystemDefinedAuthenticators(
+            List<LocalAuthenticatorConfig> systemAuthenticators) {
+
+        List<LocalAuthenticatorConfig> mergedLocalAuthenticators = new ArrayList<>();
+        Map<String, LocalAuthenticatorConfig> localAuthenticatorsInDB = new HashMap<>();
+        for (LocalAuthenticatorConfig localAuthenticator : systemAuthenticators) {
+            localAuthenticatorsInDB.put(localAuthenticator.getName(), localAuthenticator);
+        }
+
+        for (LocalAuthenticatorConfig config : localAuthenticators) {
+            if (localAuthenticatorsInDB.containsKey(config.getName())) {
+                LocalAuthenticatorConfig localAuthenticatorConfig = localAuthenticatorsInDB.get(config.getName());
+                String amrValue = localAuthenticatorConfig.getAmrValue() != null ?
+                        localAuthenticatorConfig.getAmrValue() : config.getAmrValue();
+                config.setAmrValue(amrValue);
+                mergedLocalAuthenticators.add(config);
+            } else {
+                mergedLocalAuthenticators.add(config);
+            }
```

Comment:
if-else can be simplified. Remove else part and move `mergedLocalAuthenticators.add(config);` to outside of if block